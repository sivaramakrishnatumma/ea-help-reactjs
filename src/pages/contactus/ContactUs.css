@media (min-width: 992px) {
  .page_title {
    background-color: #f5f5f5;
    display: flex;
    padding: 1rem 5rem !important;
    align-content: center;
    font-size: 2.5em !important;
    font-weight: 600;
  }
}

.page_title {
  background-color: #f5f5f5;
  display: flex;
  padding: 1rem 2rem;
  align-content: center;
  font-size: 2rem;
  font-weight: 600;
}

@media (min-width: 992px) {
  .product_container {
    max-width: 1120px !important;
  }
}

.product_container {
  padding-top: 25px;
  margin: auto;
  max-width: 90%;
}

.disclaimer {
  color: #585858;
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  margin-bottom: 16px;
  padding-right: 0;
}

.help-text {
  font-size: 24px;
  font-weight: 400;
  line-height: 18px;
  color: #161616;
  padding-bottom: 14px;
}

.search-bar {
  margin: 10px 0px;
}

.submit-button-container {
  margin: 2rem 0;
}

.loader {
  position: fixed;
  /* align-content: center; */
  top: 50%;
  left: 50%;
}

.alert {
  position: fixed !important;
  top: 20rem;
  width: 60%;
  left: 20%;
  height: 200px;
  font-size: 3rem;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 5%;
}

.close {
  position: absolute;
  right: 17px;
  top: 8px;
  font-size: 2rem;
  cursor: pointer;
}