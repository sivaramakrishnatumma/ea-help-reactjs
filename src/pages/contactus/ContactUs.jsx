import React, { Component } from "react";
import { connect } from "react-redux";
import Loader from "react-loader-spinner";

import {
  getProducts,
  resetToDefault,
  changePlatform,
  productSelected,
  platformSelected,
  topicSelected,
  issueSelected
} from "../../actions/actions";

import Breadcrumb from "../../shared/components/Breadcrumb/Breadcrumb";
import SearchBar from "../../shared/components/SearchBar/SearchBar";

import ImageGrid from "./Components/ImageGrid";
import SelectedItem from "./Components/SelectedItem";
import GridsCard from "./Components/GridsCard";
import EmailForm from "./Components/EmailForm";
import "./ContactUs.css";

class ContactUs extends Component {
  state = {
    searchKey: ""
  };

  componentDidMount() {
    this.props.dispatch(getProducts());
  }

  filterProducts = product => {
    return (
      product.altNames
        .toLowerCase()
        .indexOf(this.state.searchKey.toLowerCase()) > -1
    );
  };

  onSearch = key => {
    this.setState({ searchKey: key });
  };

  render() {
    const {
      product,
      products,
      platform,
      platforms,
      topic,
      topics,
      issue,
      issues,
      loading,
      isEmailSent,
      dispatch
    } = this.props;
    const parentPaths = [{ name: "Home", path: "/" }];
    return (
      <React.Fragment>
        <Breadcrumb items={parentPaths} currentPage="Case Information" />
        <div className="page_title">CASE INFORMATION</div>
        <div className="product_container">
          <div className="disclaimer">*Indicates required field</div>
          <div className="help-text">What can we help you with?</div>
          <div className="help-text">Select the game or service.*</div>
          {!product && (
            <SearchBar
              onKeyPress={this.onSearch}
              placeholder="Search any EA product"
              className="search-bar"
            />
          )}

          {!product && (
            <ImageGrid
              items={
                this.state.searchKey
                  ? products.filter(this.filterProducts)
                  : products
              }
              title="product"
              onSelect={product => {
                this.setState({ searchKey: "" });
                dispatch(productSelected(product));
              }}
            />
          )}

          {product && (
            <React.Fragment>
              <SelectedItem
                name={product.name}
                imageUrl={product.imageUrl}
                type="product"
                onChange={() => dispatch(resetToDefault())}
              />

              <div className="platforms-label">
                What platform are you playing on?*
              </div>
            </React.Fragment>
          )}

          {product && !platform && platforms && (
            <ImageGrid
              items={platforms}
              title="platform"
              onSelect={platform =>
                dispatch(platformSelected(platform, this.props.product.issues))
              }
            />
          )}

          {product && platform && (
            <SelectedItem
              name={platform.name}
              imageUrl={platform.imageUrl}
              type="platform"
              onChange={() => dispatch(changePlatform())}
            />
          )}

          {product && platform && (
            <GridsCard
              items={topics}
              title="topic"
              selectedItem={topic}
              onSelect={topic => dispatch(topicSelected(topic))}
            />
          )}

          {product && platform && topic && (
            <GridsCard
              items={issues}
              title="issue"
              selectedItem={issue}
              onSelect={issue => dispatch(issueSelected(issue))}
            />
          )}

          {product && platform && topic && issue && (
            <div className="submit-button-container">
              <EmailForm></EmailForm>
            </div>
          )}
        </div>
        {loading && (
          <Loader
            type="ThreeDots"
            color="#00BFFF"
            height={80}
            width={80}
            className="loader"
            //  timeout={3000} //3 secs
          />
        )}
        {isEmailSent && (
          <div className="alert alert-success" role="alert">
            <span
              className="close"
              onClick={() => {
                window.scrollTo(0, 0);
                dispatch(resetToDefault());
              }}
            >
              &times;
            </span>
            Email sent successfully
          </div>
        )}
      </React.Fragment>
    );
  }
}

const mapStateToProps = state => {
  return {
    ...state,
    product: state.selectedProduct,
    platform: state.selectedPlatform,
    topic: state.selectedTopic,
    issue: state.selectedIssue
  };
};

export default connect(mapStateToProps)(ContactUs);
