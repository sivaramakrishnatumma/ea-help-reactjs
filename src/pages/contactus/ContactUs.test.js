import { mount, shallow } from "enzyme";
import React from "react";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import ContactUs from "./ContactUs";
import { BrowserRouter } from "react-router-dom";

describe("Contact US test case", () => {
  const mockStore = configureMockStore();
  let store = mockStore({});
  const initialState = {
    products: [],
    platforms: [],
    topics: [],
    issues: [],
    product: null,
    platform: null,
    topic: null,
    issue: null
  };

  it("should test contact us page with default values", () => {
    store = mockStore(initialState);
    const wrapper = shallow(
      <Provider store={store}>
        <ContactUs></ContactUs>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should test email success alert close", () => {
    const state = { ...initialState, isEmailSent: true };
    store = mockStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ContactUs></ContactUs>
        </BrowserRouter>
      </Provider>
    );

    wrapper
      .find(".close")
      .at(0)
      .simulate("click");
    expect(wrapper).toMatchSnapshot();
  });

  it("should test products filter on search input change", () => {
    const state = { ...initialState, isEmailSent: true };
    store = mockStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ContactUs></ContactUs>
        </BrowserRouter>
      </Provider>
    );
    const event = { target: { name: "nfs" } };
    expect(wrapper).toMatchSnapshot();
    wrapper
      .find(".searchbar-input")
      .at(0)
      .simulate("change", event);
  });

  it("should test contact us page with product and platforms", () => {
    const state = { ...initialState, product: {}, platforms: [] };
    store = mockStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ContactUs></ContactUs>
        </BrowserRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
