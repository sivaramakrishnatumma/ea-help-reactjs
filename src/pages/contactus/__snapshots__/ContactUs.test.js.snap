// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact US test case should test contact us page with default values 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(ContactUs) />
</ContextProvider>
`;

exports[`Contact US test case should test contact us page with product and platforms 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(ContactUs)>
        <ContactUs
          dispatch={[Function]}
          issues={Array []}
          platforms={Array []}
          products={Array []}
          topics={Array []}
        >
          <Breadcrumb
            currentPage="Case Information"
            items={
              Array [
                Object {
                  "name": "Home",
                  "path": "/",
                },
              ]
            }
          >
            <nav
              aria-label="breadcrumb"
            >
              <ol
                className="breadcrumb"
              >
                <li
                  className="breadcrumb-item"
                  key="0"
                >
                  <Link
                    to="/"
                  >
                    <LinkAnchor
                      href="/"
                      navigate={[Function]}
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        Home
                      </a>
                    </LinkAnchor>
                  </Link>
                </li>
                <li
                  aria-current="page"
                  className="breadcrumb-item active"
                >
                  Case Information
                </li>
              </ol>
            </nav>
          </Breadcrumb>
          <div
            className="page_title"
          >
            CASE INFORMATION
          </div>
          <div
            className="product_container"
          >
            <div
              className="disclaimer"
            >
              *Indicates required field
            </div>
            <div
              className="help-text"
            >
              What can we help you with?
            </div>
            <div
              className="help-text"
            >
              Select the game or service.*
            </div>
            <SearchBar
              className="search-bar"
              onKeyPress={[Function]}
              placeholder="Search any EA product"
            >
              <div
                className="searchbar-container search-bar"
              >
                <input
                  aria-label="Search"
                  className="form-control mr-sm-2 searchbar-input"
                  onChange={[Function]}
                  placeholder="Search any EA product"
                  type="search"
                />
                <i
                  className="fa fa-search"
                />
              </div>
            </SearchBar>
            <ImageGrid
              items={Array []}
              onSelect={[Function]}
              title="product"
            >
              <div
                className="images-container"
              >
                <div
                  className="images-grid"
                />
              </div>
            </ImageGrid>
          </div>
        </ContactUs>
      </Connect(ContactUs)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Contact US test case should test email success alert close 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(ContactUs)>
        <ContactUs
          dispatch={[Function]}
          isEmailSent={true}
          issues={Array []}
          platforms={Array []}
          products={Array []}
          topics={Array []}
        >
          <Breadcrumb
            currentPage="Case Information"
            items={
              Array [
                Object {
                  "name": "Home",
                  "path": "/",
                },
              ]
            }
          >
            <nav
              aria-label="breadcrumb"
            >
              <ol
                className="breadcrumb"
              >
                <li
                  className="breadcrumb-item"
                  key="0"
                >
                  <Link
                    to="/"
                  >
                    <LinkAnchor
                      href="/"
                      navigate={[Function]}
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        Home
                      </a>
                    </LinkAnchor>
                  </Link>
                </li>
                <li
                  aria-current="page"
                  className="breadcrumb-item active"
                >
                  Case Information
                </li>
              </ol>
            </nav>
          </Breadcrumb>
          <div
            className="page_title"
          >
            CASE INFORMATION
          </div>
          <div
            className="product_container"
          >
            <div
              className="disclaimer"
            >
              *Indicates required field
            </div>
            <div
              className="help-text"
            >
              What can we help you with?
            </div>
            <div
              className="help-text"
            >
              Select the game or service.*
            </div>
            <SearchBar
              className="search-bar"
              onKeyPress={[Function]}
              placeholder="Search any EA product"
            >
              <div
                className="searchbar-container search-bar"
              >
                <input
                  aria-label="Search"
                  className="form-control mr-sm-2 searchbar-input"
                  onChange={[Function]}
                  placeholder="Search any EA product"
                  type="search"
                />
                <i
                  className="fa fa-search"
                />
              </div>
            </SearchBar>
            <ImageGrid
              items={Array []}
              onSelect={[Function]}
              title="product"
            >
              <div
                className="images-container"
              >
                <div
                  className="images-grid"
                />
              </div>
            </ImageGrid>
          </div>
          <div
            className="alert alert-success"
            role="alert"
          >
            <span
              className="close"
              onClick={[Function]}
            >
              Ã—
            </span>
            Email sent successfully
          </div>
        </ContactUs>
      </Connect(ContactUs)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Contact US test case should test products filter on search input change 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(ContactUs)>
        <ContactUs
          dispatch={[Function]}
          isEmailSent={true}
          issues={Array []}
          platforms={Array []}
          products={Array []}
          topics={Array []}
        >
          <Breadcrumb
            currentPage="Case Information"
            items={
              Array [
                Object {
                  "name": "Home",
                  "path": "/",
                },
              ]
            }
          >
            <nav
              aria-label="breadcrumb"
            >
              <ol
                className="breadcrumb"
              >
                <li
                  className="breadcrumb-item"
                  key="0"
                >
                  <Link
                    to="/"
                  >
                    <LinkAnchor
                      href="/"
                      navigate={[Function]}
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        Home
                      </a>
                    </LinkAnchor>
                  </Link>
                </li>
                <li
                  aria-current="page"
                  className="breadcrumb-item active"
                >
                  Case Information
                </li>
              </ol>
            </nav>
          </Breadcrumb>
          <div
            className="page_title"
          >
            CASE INFORMATION
          </div>
          <div
            className="product_container"
          >
            <div
              className="disclaimer"
            >
              *Indicates required field
            </div>
            <div
              className="help-text"
            >
              What can we help you with?
            </div>
            <div
              className="help-text"
            >
              Select the game or service.*
            </div>
            <SearchBar
              className="search-bar"
              onKeyPress={[Function]}
              placeholder="Search any EA product"
            >
              <div
                className="searchbar-container search-bar"
              >
                <input
                  aria-label="Search"
                  className="form-control mr-sm-2 searchbar-input"
                  onChange={[Function]}
                  placeholder="Search any EA product"
                  type="search"
                />
                <i
                  className="fa fa-search"
                />
              </div>
            </SearchBar>
            <ImageGrid
              items={Array []}
              onSelect={[Function]}
              title="product"
            >
              <div
                className="images-container"
              >
                <div
                  className="images-grid"
                />
              </div>
            </ImageGrid>
          </div>
          <div
            className="alert alert-success"
            role="alert"
          >
            <span
              className="close"
              onClick={[Function]}
            >
              Ã—
            </span>
            Email sent successfully
          </div>
        </ContactUs>
      </Connect(ContactUs)>
    </Router>
  </BrowserRouter>
</Provider>
`;
